client/src/nv_ingest_client/client/client.py

```diff
@@ -475,3 +570,99 @@ def submit_job_async(self, job_indices: Union[str, List[str]], job_queue_id: str
+        """
+        Create and submit job specifications (JobSpecs) for a batch of files, returning the job IDs.
+        This function takes a batch of files, processes each file to extract its content and type,
+        creates a job specification (JobSpec) for each file, and adds tasks from the provided task
+        list. It then submits the jobs to the client and collects their job IDs.
+
+        Parameters
+        ----------
+        files_batch : List[str]
+            A list of file paths to be processed. Each file is assumed to be in a format compatible
+            with the `extract_file_content` function, which extracts the file's content and type.
+        tasks : Dict[str, Any]
+            A dictionary of tasks to be added to each job. The keys represent task names, and the
+            values represent task specifications or configurations. Standard tasks include "split",
+            "extract", "store", "caption", "dedup", "filter", "embed", and "vdb_upload".
+
+        Returns
+        -------
+        Tuple[List[JobSpec], List[str]]
+            A Tuple containing the list of JobSpecs and list of job IDs corresponding to the submitted jobs.
+            Each job ID is returned by the client's `add_job` method.
+
+        Raises
+        ------
+        ValueError
+            If there is an error extracting the file content or type from any of the files, a
+            ValueError will be logged, and the corresponding file will be skipped.
+
+        Notes
+        -----
+        - The function assumes that a utility function `extract_file_content` is defined elsewhere,
+          which extracts the content and type from the provided file paths.
+        - For each file, a `JobSpec` is created with relevant metadata, including document type and
+          file content. Various tasks are conditionally added based on the provided `tasks` dictionary.
+        - The job specification includes tracing options with a timestamp (in nanoseconds) for
+          diagnostic purposes.
+
+        Examples
+        --------
+        Suppose you have a batch of files and tasks to process:
+        >>> files_batch = ["file1.txt", "file2.pdf"]
+        >>> tasks = {"split": ..., "extract_txt": ..., "store": ...}
+        >>> client = NvIngestClient()
+        >>> job_ids = client.create_job_specs_for_batch(files_batch, tasks)
+        >>> print(job_ids)
+        ['job_12345', 'job_67890']
+
+        In this example, jobs are created and submitted for the files in `files_batch`, with the
+        tasks in `tasks` being added to each job specification. The returned job IDs are then
+        printed.
+
+        See Also
+        --------
+        create_job_specs_for_batch: Function that creates job specifications for a batch of files.
+        JobSpec : The class representing a job specification.
+        """
+        job_specs = create_job_specs_for_batch(files_batch)
+
+        job_ids = []
+        for job_spec in job_specs:
```

Comment:
While we're in process of turning the CLI code into a library, I'd like to be a bit more precise here and handle corner cases where more than one of a single task type is requested... or if multiple tasks of the same type are selected with different configuration parameters.

Its probably ok if we just reject duplicate tasks out of hand for now and raise an error, but its also worth thinking through when/if we might want them.

Reply:
Thanks. I thought about this, but I opted to reject duplicate task immediately and raise an error if there are any duplicate tasks in [e1fceb6](https://github.com/NVIDIA/nv-ingest/pull/158/commits/e1fceb6c1ff9750b7eb2201de80bc1b66760a2bb), mostly to avoid complexity.

Given the serial nature of our current pipeline, I didn't think duplicate tasks made much sense, but maybe there are some use cases where users might want to apply, for example, split tasks to pdf documents but not on pptx documents, or something like that? Or maybe in that case, two separate pipelines makes more sense.